sdk/python/feast/embedded_go/type_map.py

```diff
@@ -50,14 +50,14 @@
     PrimitiveFeastType.BYTES: pa.binary(),
     PrimitiveFeastType.BOOL: pa.bool_(),
     PrimitiveFeastType.UNIX_TIMESTAMP: pa.timestamp("s"),
-    Array(PrimitiveFeastType.INT32): pa.list_(pa.int32()),
-    Array(PrimitiveFeastType.INT64): pa.list_(pa.int64()),
-    Array(PrimitiveFeastType.FLOAT32): pa.list_(pa.float32()),
-    Array(PrimitiveFeastType.FLOAT64): pa.list_(pa.float64()),
-    Array(PrimitiveFeastType.STRING): pa.list_(pa.string()),
-    Array(PrimitiveFeastType.BYTES): pa.list_(pa.binary()),
-    Array(PrimitiveFeastType.BOOL): pa.list_(pa.bool_()),
-    Array(PrimitiveFeastType.UNIX_TIMESTAMP): pa.list_(pa.timestamp("s")),
+    Array(base_type=PrimitiveFeastType.INT32): pa.list_(pa.int32()),
```

Comment:
Ah, this was a mistake, thanks for catching it! If you look in feast/types.py, Array inherits from ComplexFeastType, which inherits from BaseModel, so I had to change the Array constructor because Pydantic models don't allow assignment in the init method. Initially I thought I would have to use a keyword argument for base_type everywhere, before realizing I could do this instead:

    def __init__(self, base_type: Union[PrimitiveFeastType, ComplexFeastType]):
        super(Array, self).__init__(base_type=base_type)
        
        
But I forgot to remove one of the things I'd changed. Fixed.

Reply:
I built off of @wparsley-eg's changes, so I'll let him explain